savvy_scheduler.md

```diff
@@ -82,98 +77,33 @@ The second part of the lab is to set up scheduler itself and manage incoming job
 
 The last part of your job is computing stats and clean-up, which is fairly trivial. You may need some extra variables to help you keep track of these stats.
 
-
-## Job Struct
-
-We have provided a job struct defined in libscheduler.h. You do not need to modify the __state__, __ctx__, or __stack_start__ fields. The only field you will be using or modifying is __metadata__, where you must insert your __job_info__ struct that you will define in __libscheduler.c__.
-
-## Functions you need to implement
-
-The only graded file is __libscheduler.c__. You will need to augment the `job_info` struct and implement the following functions:
-
-### void scheduler_start_up(scheme_t s)
-
-This function is implemented for you, but you can add some code to it if you need to initialize any global variables.
-
-### int comparer_fcfs(const void \*a, const void \*b)
-
-Comparer function for fcfs scheme
-
-### int comparer_ppri(const void \*a, const void \*b) 
-
-Comparer function for ppri scheme
-
-### int comparer_pri(const void \*a, const void \*b) 
-
-Comparer function for pri sscheme
-
-### int comparer_psrtf(const void \*a, const void \*b)
-
-Comparer function for psrtf scheme
-
-### int comparer_rr(const void \*a, const void \*b) 
-
-Comparer function for rr scheme
-
-### int comparer_sjf(const void \*a, const void \*b) 
-
-Comparer function for sjf scheme
-
-### void scheduler_new_job(job \*newjob, int job_number, double time, scheduler_info \*sched_data)
```

Comment:
IMO understanding the use of the new_job, quantum_expired, and job_finished functions was the most confusing part of this lab. Unless I'm missing where you added this back elsewhere, I feel like having this documentation is useful here.

Reply:
I plan to add this into the `.h` files. Comments like these should all be in the header files, and not split up between the docs and the `.h` files

Reply:
Good point, definitely agree.